src/client_lib/pegasus_scanner_impl.cpp

```diff
@@ -162,13 +168,16 @@ void pegasus_client_impl::pegasus_scanner_impl::_async_next_internal()
         }
 
         // valid data got
-        std::string hash_key, sort_key;
-        pegasus_restore_key(_kvs[_p].key, hash_key, sort_key);
-        std::string value(_kvs[_p].value.data(), _kvs[_p].value.length());
-        uint32_t expire_ts_seconds = _kvs[_p].__isset.expire_ts_seconds
-                                         ? static_cast<uint32_t>(_kvs[_p].expire_ts_seconds)
-                                         : 0;
-
+        std::string hash_key, sort_key, value;
+        uint32_t expire_ts_seconds = 0;
+        // _kv_count == -1 means req just want to get data counts, not include data value
+        if (_kv_count == -1) {
```

Comment:
should be _kv_count != -1 ?

Reply:
You should explain the case of `_kv_count == -1` and all other cases, `_kv_count > -1` is only part of case you have to explain.

Reply:
How about add another variable to indicate directly what you want to do?